xml-occurrence-parsers/src/main/java/org/gbif/xml/occurrence/parser/identifier/OccurrenceKeyHelper.java

```diff
@@ -60,20 +60,35 @@ public static String buildUnscopedKey(@Nullable PublisherProvidedUniqueIdentifie
 
   @Nullable
   public static String buildUnscopedKey(@Nullable HolyTriplet triplet) {
-    if (triplet == null
-        || triplet.getDatasetKey() == null
-        || triplet.getInstitutionCode() == null
-        || triplet.getCollectionCode() == null
-        || triplet.getCatalogNumber() == null) {
+    if (Objects.isNull(triplet)
+        || Objects.isNull(triplet.getDatasetKey())
```

Comment:
maybe datasetKey is not needed here?

Reply:
I suggest refactoring some old parts in a separated task